### PR TITLE
Search functionality improvement

### PR DESCRIPTION
**Pull request by NimSum**
Search now finds each individual matching word for dataset titles

### CODE REVIEW

**Comment by raechelo** [comment]
Nice job getting the search function to be better and more intuitive for user interaction! I think this is exactly what we needed to have!

**Comment by alizarincrimson** [comment]
Great refactor Nim, this search functionality is 100% how I would expect it to behave as a user!

File: src/components/Search.jsx [hunk 1]
```diff
@@ -11,20 +11,29 @@ export class Search extends Component {
     this.setState({ value: e.target.value })
   }
 
+  rmSpecChars(word) {
+    return word.split('').filter((ltr) => {
+      return !["'", " ", "-", ",", ":", "#", ".", "!", "?"].includes(ltr)
+    }).join('')
```
Comment (raechelo) [inline src/components/Search.jsx:17]: nice job splitting this method up, i think it was getting too long before. good thinking so we can follow SRP.

File: src/components/Search.jsx [hunk 2]
```diff
@@ -11,20 +11,29 @@ export class Search extends Component {
...
-    }).join('')
-    console.log(userInput)
-    let a = this.props.data.filter((item) => {
-      return userInput.match(item.title.toUpperCase().split('').filter((ltr) => {
-        return !specChars.includes(ltr)
-      }).join('')) ?
-      item : false
-    })
-    console.log(a)
+    let userInput = this.state.value.split(' ').length > 1
```
Comment (raechelo) [inline src/components/Search.jsx:32]: i think this looks so much better than how we had it before. our handlesubmit now only really needs to do one thing which is super!

**Pull request merged**